### PR TITLE
Refactor of webview code

### PR DESCRIPTION
**Pull request by rudyflores**
## Proposed changes

Refactor of existing Webview code to be easily extensible, this also integrates a watcher that will hot reload your changes on your webview to the debugger so you can see your changes just by saving!

To add new views in the future just:

1. add new folder with webview name to `zowe-explorer/src/webviews/src`.
2. add the following components: `index.html`, `app.tsx`, `index.tsx`, `types.ts` (optional)

And that's it! 

## Release Notes

Milestone: 2.12.0

Changelog: Added new method of creating webviews on Zowe Explorer

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [x] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [x] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

**Comment by rudyflores** [comment]
FYI: the code smell is from existing `App.tsx` work from @traeok 's work on editing attributes, this can be perhaps be addressed in a separate PR since the code in that file could be restructures into components in the future.

File: packages/zowe-explorer-api/src/vscode/ui/WebView.ts [hunk 1]
```diff
@@ -39,11 +39,16 @@ export class WebView {
      * The webview entrypoint must be located at <webview folder>/dist/assets/index.js.
      *
      * @param title The title for the new webview
-     * @param dirName The directory name (relative to the "webviews" folder in the extension root) with a valid entrypoint (see above).
+     * @param webviewName The directory name (relative to the "webviews" folder in the extension root) with a valid entrypoint (see above).
```
Comment (awharn) [inline packages/zowe-explorer-api/src/vscode/ui/WebView.ts:42]: The description for this parameter does not match the parameter name

**Review by traeok — APPROVED** [review]
LGTM, thanks @rudyflores!

**Review by awharn — DISMISSED** [review]

**Review by traeok — CHANGES_REQUESTED** [review]
Sorry for the sudden change in review @rudyflores. I was under the impression that the dev host was showing the latest changes, but the `watch` script seems to be failing for me:

image

**Review by zFernand0 — DISMISSED** [review]

**Comment by rudyflores** [comment]
All fixed now @traeok good catch on the windows end of things for concurrently!

**Review by traeok — APPROVED** [review]

**Review by zFernand0 — APPROVED** [review]

**Pull request merged**